src/Common/Core/Impl/Disposables/Disposable.cs

```diff
@@ -25,6 +25,20 @@ public static class Disposable {
         }
 
         /// <summary>
+        /// Creates a disposable wrapper for a disposable that will be invoked at most once.
+        /// </summary>
+        /// <param name="disposable">Disposable that will be wrapped. Disposable is guaranteed to be run at most once.</param>
+        /// <returns>The disposable object that disposes wrapped object.</returns>
+        /// <exception cref="T:System.ArgumentNullException"><paramref name="disposable" /> is null.</exception>
+        public static IDisposable Create(IDisposable disposable) {
+            if (disposable == null) {
```

Comment:
You can use `Check.ArgumentNull(nameof(disposable), disposable);` for brevity if you wish